docs/source/notes/serialization.rst

```diff
@@ -2,90 +2,317 @@
+
+::
+
+    >>> t = torch.tensor([1., 2.])
+    >>> torch.save(t, 'tensor.pt')
+    >>> torch.load('tensor.pt')
+    tensor([1., 2.])
+
+By convention, PyTorch files are typically written with a ‘.pt’ or ‘.pth’ extension.
+
+:func:`torch.save` and :func:`torch.load` use Python’s pickle by default,
+so you can also save multiple tensors as part of Python objects like tuples,
+lists, and dicts:
+
+::
+
+    >>> d = {'a': torch.tensor([1., 2.]), 'b': torch.tensor([3., 4.])}
+    >>> torch.save(d, 'tensor_dict.pt')
+    >>> torch.load('tensor_dict.pt')
+    {'a': tensor([1., 2.]), 'b': tensor([3., 4.])}
+
+Custom data structures that include PyTorch tensors can also be saved if the
+data structure is pickle-able.
+
+Saving and loading tensors preserves views
 ---------------------------------------------
 
 .. _preserve-storage-sharing:
 
-PyTorch saves the underlying storages so that tensors sharing the same storage before :func:`torch.save`
-will still share storage after :func:`torch.load`.
+Saving tensors preserves their view relationships:
+
+::
+
+    >>> numbers = torch.arange(1, 10)
+    >>> evens = numbers[1::2]
+    >>> torch.save([numbers, events], 'tensors.pt')
+    >>> loaded_numbers, loaded_evens = torch.load('tensors.pt')
+    >>> loaded_evens *= 2
+    >>> loaded_numbers
+    tensor([ 1,  4,  3,  8,  5, 12,  7, 16,  9])
+
+`loaded_numbers` and `loaded_evens` share storage just like `numbers` and
+`evens` do.
+
+Behind the scenes, PyTorch saves storage objects and tensor metadata separately,
+and it reuses the storage object where possible. In the above case, for example,
+the values [1, 2, 3, 4, 5, 6, 7, 8, 9] are only saved once. This typically saves
+space, but if small views of a much larger storage are saved then size of
+the saved file may become prohibitively large, as in the following example:
 
 ::
 
-    >>> tensor = torch.zeros(1000000)
-    >>> slice1 = tensor[:1000]
-    >>> slice2 = tensor[:10] # slice1 and slice2 share the same storage
-    >>> torch.save([slice1, slice2], 'share.pt')
-    >>> loaded_1, loaded_2 = torch.load('share.pt')
-    >>> loaded_1[0]
-    tensor(0.)
-    >>> loaded_2[0]
-    tensor(0.)
-    >>> loaded_2[0] = 1
-    >>> loaded_1[0] # loaded tensors still share storage
-    tensor(1.)
-
-Note that saving storage instead of tensor itself means the serialized file size might not match tensor size.
-In the example above the whole `tensor`'s storage (of size 1000000) is serialized instead of only slices.
-When tensor is expanded from a smaller storage, serialized file size might be smaller than tensor size as well.
+    >>> numbers = torch.arange(1, 1e5)
+    >>> one_hand = numbers[0:5]
+    >>> two_hands = numbers[5:10]
+    >>> torch.save([one_hand, two_hands], 'tensors.pt')
+
+
+`one_hand` and `two_hands` are both (tiny) views of `numbers`, and when saved
+their storage, `numbers`, is saved. So instead of saving ten numbers
+([1, 2, 3, 4, 5], [6, 7, 8, 9, 10]) PyTorch will save 99,999!
```

Comment:
maybe show how to find this value, so people can check it themselves (something like `one_hand.storage().size`.